### PR TITLE
planner: support foreign key choose parent schema as default in CreateTable stmt 

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Aligned with MySQL behavior
```
drop database if exists fk_test;
create database if not exists fk_test;
create table fk_test.t1 (c1 int not null primary key);
create table fk_test.t2 (id int not null primary key, c1 int not null, constraint foreign key (c1) references t1(c1));
```
MySQL can deal with above sql well, but TiDB will get `ERROR 1046 (3D000) at line 4: No database selected`

### What is changed and how it works?
if foreign key has no schema, use parent table schema

### Check List 

Tests 
 - Unit test

Code changes
 - NA

Side effects
 - NA

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Fix a compatibility issue that if the foreign key constraint in `CREATE TABLE` statement has no schema, schema of the created table should be used instead of returning `No Database selected` error.

### CODE REVIEW

**Comment by zz-jason** [comment]
@3pointer please add some proper labels.

**Comment by 3pointer** [comment]
> Please add a test case for this PR, thanks!

Sure, add test [here](https://github.com/pingcap/tidb/pull/12548/files#diff-703ae6b7872b425273d1832c198598c8R2086)

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -2082,3 +2082,14 @@ func (s *testIntegrationSuite3) TestSqlFunctionsInGeneratedColumns(c *C) {
 	tk.MustGetErrCode("create table t (a int, b int as (statement_digest(1)))", mysql.ErrGeneratedColumnFunctionIsNotAllowed)
 	tk.MustGetErrCode("create table t (a int, b int as (statement_digest_text(1)))", mysql.ErrGeneratedColumnFunctionIsNotAllowed)
 }
+
+func (s *testIntegrationSuite3) TestParserIssue284(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table test.t_parser_issue_284(c1 int not null primary key)")
```
Comment (crazycs520) [inline ddl/db_integration_test.go]: Table `t_parser_issue_284 ` created but not used? Should used when to create table `t_parser_issue_284_2 `?

Comment (3pointer) [reply ddl/db_integration_test.go]: Sure, [fixed](https://github.com/pingcap/tidb/pull/12548/files#diff-703ae6b7872b425273d1832c198598c8R2090)

**Pull request merged**